src/lib/yaraparse.py

```diff
@@ -186,44 +235,63 @@ def flatten_regex_or_tree(unit: Any) -> Optional[List[bytes]]:
         return None
 
 
+def urisfy_regex_or_tree(
+    unit: Any, is_ascii: bool, is_wide: bool, is_nocase: bool,
+) -> Optional[UrsaExpression]:
+    or_strings = flatten_regex_or_tree(unit)
+    if or_strings and all(s is not None for s in or_strings):
+        or_ursa_strings: List[UrsaExpression] = []
+        for s in or_strings:
+            ursified_s = ursify_regex_text(
+                s, is_ascii=is_ascii, is_wide=is_wide, is_nocase=is_nocase
+            )
+            if ursified_s:
+                or_ursa_strings.append(ursified_s)
+        if or_ursa_strings:
+            return UrsaExpression.or_(*or_ursa_strings)
+        else:
+            return None
+    else:
+        return None
+
+
 def urisfy_regex(
-    units: List[Any],
+    regex: Any,
     is_ascii: bool = False,
     is_wide: bool = False,
     is_nocase: bool = False,
 ) -> Optional[UrsaExpression]:
     strings: List[UrsaExpression] = []
 
-    joined_string = b""
-    for i, unit in enumerate(units):
-        if type(unit) is RegexpText:
-            joined_string += unit.text.encode()
-        elif type(unit) is RegexpGroup:
-            or_strings = flatten_regex_or_tree(unit.unit)
-            if or_strings and all(s is not None for s in or_strings):
-                or_ursa_strings = [
-                    ursify_plain_string(
-                        s,
-                        is_ascii=is_ascii,
-                        is_wide=is_wide,
-                        is_nocase=is_nocase,
-                    )
-                    for s in or_strings
-                ]
-                strings.append(UrsaExpression.or_(*or_ursa_strings))
-
-        if joined_string and (
-            type(unit) is not RegexpText or i == len(units) - 1
-        ):
-            strings.append(
-                ursify_plain_string(
+    if type(regex) is not RegexpConcat:
```

Comment:
What do you think about reversing the if statement here? i.e. converting the other branch to `if type(regex) is RegexpConcat` and then this one to `else`